first commit reduce7 in typescript

d-flytech opened the pull request:
Learning how to reduce in typescript.

FredDukker commented on reduce7/reduce7.ts:
```diff
@@ -0,0 +1,27 @@
+// Defines Gender as a union type; an enum could also be used in more complex scripts
```
Looks good.
Did you think about when to use to type and when interface?

d-flytech replied:
You can use types for unions, primitives and utility patterns and interfaces you can use for extensibility, for objects and classes.

chimurai commented on reduce7/reduce7.ts:
```diff
@@ -0,0 +1,27 @@
+// Defines Gender as a union type; an enum could also be used in more complex scripts
+type Gender = "male" | "female" | "x";
```
Can you make `type Gender` and `interface Person` reusable?
Extract these two into a new file `types.ts` and use them in this and previous `.ts` implementations.

d-flytech replied:
Yes , I will do that, but I don't know where to place the types.ts. I'am not allowed to work in main? If I place it in every branch you will have redundency. I can place the types.ts in a seperate branch and merge it to the other branches. Later after review I can merge it to main. Is that logical?

chimurai replied:
You can add `types.ts` in the root of this branch. You should be able to add the types to the previous tasks.
If the folders are outdated, you can merge `main` branch into this branch again.

d-flytech commented:
I updated reduce7 with types.ts from the root data-structure and changed the reduce7.ts script and committed them both in reduce7 branch. If I do this with the rest of the branches I will get redundancy. I prefer to commit types.ts in main or a seperate branch to avoid this. I am strungeling in what is expected from me. I would like to know what I can do best before I change the other assignments. Please advise!

chimurai commented:
This update looks good and is good to merge.

Could you create a new branch and implement your new type in the previous typescript files?
